src/main/java/duke/util/Program.java

```diff
@@ -0,0 +1,249 @@
+            deleteTask(string);
+        } else {
+            throw new HalException("I'm sorry, but I don't know what that means :((");
+        }
+        UserData.writeToFile(parser.saveListAsString(listTasks));
+    }
+
+    //function to load tasks from memory
+    public static void loadSavedTasks() throws IOException {
+        listTasks = readFromFile();
+        int count = 0;
+        for (Object obj : listTasks) {
+            if ( obj != null ) {
+                count++;
+            }
+        }
+        numItems = count;
+    }
+
+    //function to add a new deadline task
+    public static void addDeadlineTask(String deadlineTask) throws HalException {
+        System.out.println(LINE_BREAK_SINGLE);
+
+        //check if string contains '/by' tag
+        if (!deadlineTask.contains("/by")) {
+            System.out.println(PRINT_ERROR_MESSAGE);
+            throw new HalException("Wrong Deadline task format");
+        }
+
+        String description = deadlineTask.substring(DEADLINE_INDEX, deadlineTask.indexOf('/')).trim();
+        String by = deadlineTask.substring(deadlineTask.indexOf("/by") + TASK_STRING_OFFSET).trim();
+
+        if (description.equals("")) {
+            System.out.println(PRINT_EMPTY_DESCRIPTION_MESSAGE);
+            throw new HalException("Empty description");
+        } else if (by.equals("")) {
+            System.out.println(PRINT_EMPTY_DATE_MESSAGE);
+            throw new HalException("Empty date");
+        }
+
+        Deadline newDeadlineTask = new Deadline(description, by);
+        listTasks.add(newDeadlineTask);
+        numItems++;
+
+        System.out.println(TASK_ADDED_DONE_TEXT);
+        System.out.println(newDeadlineTask.toString());
+        printTotalTasks();
+        System.out.println(LINE_BREAK_SINGLE);
+        System.out.print(ENTER_COMMAND_TEXT);
+    }
+
+    //function to add a new event task
+    public static void addEventTask(String eventTask) throws HalException {
+        System.out.println(LINE_BREAK_SINGLE);
+
+        //check if string contains '/at' tag
+        if (!eventTask.contains("/at")) {
+            System.out.println(PRINT_ERROR_MESSAGE);
+            throw new HalException("Wrong Event task format");
+        }
+
+        String description = eventTask.substring(EVENT_INDEX, eventTask.indexOf('/')).trim();
+        String at = eventTask.substring(eventTask.indexOf("/at") + TASK_STRING_OFFSET).trim();
+
+        if (description.equals("")) {
+            System.out.println(PRINT_EMPTY_DESCRIPTION_MESSAGE);
+            throw new HalException("Empty description");
+        } else if (at.equals("")) {
+            System.out.println(PRINT_EMPTY_DATE_MESSAGE);
+            throw new HalException("Empty date");
+        }
+
+        Event newEventTask = new Event(description, at);
+        listTasks.add(newEventTask);
+        numItems++;
+
+        System.out.println(TASK_ADDED_DONE_TEXT);
+        System.out.println(newEventTask.toString());
```

Comment:
Might be good for you to check and read up on .toString() method to remove redundancies which might affect readability